### PR TITLE
Added search indexing and querying 

### PR DESCRIPTION
**Pull request by Yantrio**
This PR adds 2 components

- /search/pg-indexer - This is a tool to take our search.ndjson file and index this data into a postgres database
- /search/worker - A cloudflare worker that can query the postgres database to provide search functionality.

### CODE REVIEW

File: search/pg-indexer/main.go [hunk 1]
```diff
@@ -0,0 +1,319 @@
...
+
+	// If the most recent job is in progress, exit
+	// TODO: Handle this in a better way
+	if mostRecentJob.InProgress() {
+		log.Println("Import already in progress, exiting")
+		os.Exit(0)
+	}
+
+	scanner := bufio.NewScanner(body)
+	scanner.Scan()
```
Comment (cam72cam) [inline search/pg-indexer/main.go:61]: Check return value?

File: search/pg-indexer/main.go [hunk 2]
```diff
@@ -0,0 +1,321 @@
...
+		log.Fatal(err)
+	}
+
+	tx, err := db.Begin()
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	handled := 0
+	batchItems, err := readItems(scanner, batchSize)
```
Comment (cam72cam) [inline search/pg-indexer/main.go:96]: Shouldn't this be in a loop?

File: search/pg-indexer/main.go [hunk 3]
```diff
@@ -0,0 +1,321 @@
...
+		toDelete := make([]SearchIndexItem, 0, batchSize)
+
+		for _, item := range batchItems {
+			if item.Type == "add" {
+				toInsert = append(toInsert, item)
+				handled++
+			} else if item.Type == "deletion" {
+				toDelete = append(toDelete, item)
+				handled++
+			}
```
Comment (cam72cam) [inline search/pg-indexer/main.go:113]: What if the item type == "magicbeans"?

File: search/worker/src/query.ts [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+type Entity = {
+	id: string;
+	addr: string;
+	title: string;
+	description: string;
+	link_variables: Record<string, string>;
+};
+
+export const query = async (client: Client, queryParam: string): Promise<Entity[]> => {
+	const { rows } = await client.query(searchQuery, [queryParam]);
```
Comment (cam72cam) [inline search/worker/src/query.ts:85]: Does not handle url-encoded values, ex: `opentofu/null`

File: search/pg-indexer/schema.sql [hunk 5]
```diff
@@ -0,0 +1,21 @@
+CREATE TABLE IF NOT EXISTS entities
```
[the remark this answers is not in the document]
Comment (Yantrio) [reply search/pg-indexer/schema.sql:1]: This is just here to have as reference, I'll look into creating tables.

File: search/worker/.editorconfig [hunk 6]
```diff
@@ -0,0 +1,12 @@
+# http://editorconfig.org
```
[the remark this answers is not in the document]
Comment (Yantrio) [reply search/worker/.editorconfig:1]: its from the worker template, i can scrap it and clean this area up

**Pull request merged**